libcontainer/container_linux.go

```diff
@@ -1215,6 +1215,7 @@ func (c *linuxContainer) Restore(process *Process, criuOpts *CriuOpts) error {
 			OrphanPtsMaster: proto.Bool(true),
 			AutoDedup:       proto.Bool(criuOpts.AutoDedup),
 			LazyPages:       proto.Bool(criuOpts.LazyPages),
+			AutoExtMnt:      proto.Bool(criuOpts.AutodetectExternalMounts),
 		},
```

Comment:
Why not set AutoExtMnt by default, instead of adding a cli option (is there a case where we don't want to auto-detect of external mounts)?
LXC seems to do that and I haven't seen issues related to this, so far.

Reply:
Basically I did not want to change any existing behaviour. No idea if setting it could break something. Always setting it would be even easier for me.

Reply:
> Why not set AutoExtMnt by default, instead of adding a cli option (is there a case where we don't want to auto-detect external mounts)?

Because this option has "undefined" behavior and if you are going to migrate a container to another host, I don't recommend to use this option.

Reply:
@avagin Thanks for letting us know that you do not recommend to use it. Concerning this pull request it does not hurt or break anything, so it could be added without a problem.

I am currently using it as there seems to be a problem between checkpointing and restoring with runc. During checkpoint runc let's CRIU know that it /dev/shm (in my case) is an external bind mount (correct). In the external mount mapping I have key of /dev/shm with a value of /dev/shm. During restore runc does not tell CRIU what to do with the /dev/shm mountpoint. So CRIU fails with something like: missing mapping for /dev/shm. Autodetect external mounts fixes it (or works around it). This can/needs to be fixed probably in a different way in runc.

Reply:
First of all, we need to understand why an ext mount for /dev/shm isn't set on restore. Can you describe how I can reproduce this?

Reply:
I figured it out, it was missing from the config.json during restore. If I do it correctly it actually works :wink: 

So I do not really need this option, but we could still add it to runc for people that need it.

Reply:
@adrianreber It is the main question whether we have users who need it or not. I think this option is only for experiments and it should be never used in production, so I would recommend to not add it in runsc.

Reply:
I will remove this commit.